### ISSUE TITLE
Collect approvers-Report changed to invoice by employee still displayed in ND

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#38132]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.50-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4413252
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** applause internal team
**Slack conversation:**

## Action Performed:
1.Create a new Gmail account in OldDot (must be new)
2.Create a collect workspace
3.Set the workspace policy to USD
4.Open the JS console and run the following script in the collect policy settings page
if (Url.getCurrentPageName()!="policy" ){ alert('Only run this snippet from a policy editor!'); } else { var p = Policy.getCurrent(); p.policy.isPolicyExpenseChatEnabled = "true"; p.save().done(function(){$.jGrowl('Workspace chat creation enabled!')}); }
5.Invite another account and set it as employee - This account will be called Employee1 in the following TCs
6.Login to OldDot with the employee account
7.Navigate to workspaces and set the collect policy to default if not already set
8.Navigate to reports
9.Create a report with no expenses on them
10. Navigate to the empty report and change the report type to "Invoice"
11. As the employee in NewDot - Navigate to the workspace chat

## Expected Result:
The report that was changed to invoice is not displayed anymore in the chat

## Actual Result:
The report is still displayed in ND, if an expense is added , it shows as negative.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/43996225/adc48eee-973b-4571-83a6-f3e7e1b8a28f

![Bug6410368_1710195222125!invoiceexpence](https://github.com/Expensify/App/assets/43996225/06258be5-99ec-4559-8b3e-d770de803e76)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kavimuru** [comment]
we think this bug might be related to #wave-collect - RELEASE 1

**Comment by kavimuru** [comment]
@adelekennedy  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by trjExpensify** [comment]
I would put this one under #vip-billpay tbh. We're going to need to handle this gracefully taking into consideration invoice rooms and the likes. CC: @danielrvidal @davidcardoza

**Comment by davidcardoza** [comment]
Yeah, I think this one makes sense to fall under the @vip-billpay project. That's a pretty bad bug, I think it has to do with the transfer the object from expense report to invoice, right? Anyways I added it to the project board, let me know how I can help with this one if needed!

**Comment by adelekennedy** [comment]
@davidcardoza @trjExpensify Should we add the external label to this one or will this be all internal?

**Comment by trjExpensify** [comment]
Yeah, I'm not sure. @marcochavezf might have an idea on how we're supposed to be handling this today, technically.  

The ideal solution is that the report type `expense` gets changed to `invoice` and moves to the applicable invoice room. But we haven't built invoice rooms yet, so what is expected in the interim is a little unclear. We could be just hiding `invoice` reports from the `policyExpenseChat` report type. That is likely to get undone if the "invoice rooms" are an extension of that report type.

**Comment by adelekennedy** [comment]
@marcochavezf can you take a look here and let me know if this needs to be an internal assignment?

**Comment by marcochavezf** [comment]
Yeah, I'm going to try to look at it today to have a

**Comment by marcochavezf** [comment]
We're not changing the expense type in Onyx when the report is converted to invoice:

<img width="1153" alt="Screenshot 2024-03-15 at 3 10 02 p m" src="https://github.com/Expensify/App/assets/6829422/c1c793c1-338f-4417-9a2f-e2af9d636e31">

<img width="1087" alt="Screenshot 2024-03-15 at 3 11 05 p m" src="https://github.com/Expensify/App/assets/6829422/5ab7606b-35c8-4561-9a35-34c83855f3c2">

Independently of hiding it, I think that's something we'd need to implement

**Comment by marcochavezf** [comment]
Created a PR to send the report type to NewDot, WIP

**Comment by adelekennedy** [comment]
@marcochavezf deprioritizing to weekly

**Comment by adelekennedy** [comment]
unassigning myself as marco is working on this, it's assigned to the correct project and no payments will be due
